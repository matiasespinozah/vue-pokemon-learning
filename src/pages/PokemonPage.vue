<template>

  <h1 v-if="!pokemon"> Espere por favor </h1>

  <div v-else>
    <h1> ¿Quién es este Pokémon? </h1>

    <PokemonPicture :pokemon-id="pokemon?.id" :show-pokemon="showPokemon" />
    <PokemonOption :pokemons="pokemons" @selection="checkAnswer" />

    <template v-if="showAnswer">
      <h2 >{{ message }}</h2>
      <button @click="newGame"> Nuevo Juego </button>
    </template>
  </div>
</template>

<script>
import PokemonPicture from '@/components/PokemonPicture.vue'
import PokemonOption from '@/components/PokemonOption.vue'
import getPokemonsOptions from '@/helpers/getPokemonOptions.js'

export default {
  name: 'PokemonPage',
  components: {
    PokemonPicture,
    PokemonOption
  },
  data() {
    return {
      pokemons: [],
      pokemon: null,
      showPokemon: false,
      showAnswer: false,
      message: null
    }
  },
  methods: {
    async mixPokemons() {
      this.pokemons = await getPokemonsOptions()
      this.pokemon = this.pokemons[Math.floor(Math.random() * 4)]
    },
    checkAnswer(pokemonId) {
      this.showPokemon = true;
      this.showAnswer = true;
      if (pokemonId === this.pokemon.id) {
        this.message = `Correcto, ${this.pokemon.name}`
      } else {
        this.message = `Ops!!, era ${this.pokemon.name}`
      }
    },
    newGame() {
      this.showPokemon = false
      this.showAnswer = false
      this.pokemons = []
      this.pokemon = null
      this.mixPokemons()
    }
  },
  mounted() {
    this.mixPokemons()
  }
}
</script>
