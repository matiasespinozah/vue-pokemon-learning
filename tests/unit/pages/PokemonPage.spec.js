import {shallowMount} from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'
import { mockPokemons } from '../mocks/pokemons.mock';

describe('PokemonPage', () => {

  let wrapper = shallowMount(PokemonPage)

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage)
  })



  test('debe de hacer match con el snapshot ', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe de llamar mixPokemon al montar', () => {
    const mixPokemonSpy = jest.spyOn(PokemonPage.methods, 'mixPokemons')
    shallowMount(PokemonPage)
    expect(mixPokemonSpy).toHaveBeenCalled()
  })

  test('debe de hacer match con el snapshot cuando cargan los pokemon', () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemons: mockPokemons,
          pokemon: mockPokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: null
        }
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe de mostrar los componentes de PokemonPicture y PokemonOption ', () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemons: mockPokemons,
          pokemon: mockPokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: null
        }
      }
    })
    const picture = wrapper.find('pokemon-picture-stub')
    const option = wrapper.find('pokemon-option-stub')

    expect(picture.exists()).toBeTruthy()
    expect(option.exists()).toBeTruthy()
    expect(picture.attributes('pokemonid')).toBe('1')
    expect(option.attributes('pokemons')).toBeTruthy()
  })
  
  
  
  
})