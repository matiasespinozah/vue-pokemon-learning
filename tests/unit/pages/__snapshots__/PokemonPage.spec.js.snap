// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage debe de hacer match con el snapshot  1`] = `<h1> Espere por favor </h1>`;

exports[`PokemonPage debe de hacer match con el snapshot cuando cargan los pokemon 1`] = `
<div>
  <h1> ¿Quién es este Pokémon? </h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-option-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-option-stub>
  <!--v-if-->
</div>
`;
